src/orange/surf/Involute.hh

```diff
@@ -0,0 +1,347 @@
+ with
+ * radius \em r_b from a starting angle of \em a (\f$r/h\f$ for a finite cone.
+ */
+class Involute
+{
+  public:
+    //@{
+    //! \name Type aliases
+    using Intersections = Array<real_type, 3>;
+    using StorageSpan = Span<real_type const, 4>;
+    //@}
+
+  public:
+    //// CLASS ATTRIBUTES ////
+
+    // Surface type identifier
+    static CELER_CONSTEXPR_FUNCTION SurfaceType surface_type()
+    {
+        return SurfaceType::inv;
+    }
+
+    // Safety
+    static CELER_CONSTEXPR_FUNCTION bool simple_safety() { return false; }
+
+  public:
+    //// CONSTRUCTORS ////
+    // Construct at (0,0,z)
+    static Involute at0origin(Real3 const& origin,
+                              real_type radius,
+                              real_type a,
+                              real_type sign,
+                              real_type tmin,
+                              real_type tmax);
+
+    // Construct from origin and radius of circle of involute
+    inline CELER_FUNCTION Involute(Real3 const& origin,
+                                   real_type radius,
+                                   real_type a,
+                                   real_type sign,
+                                   real_type tmin,
+                                   real_type tmax);
+
+    // Construct from raw data
+    template<class R>
+    explicit inline CELER_FUNCTION Involute(Span<R, StorageSpan::extent>);
+
+    //// ACCESSORS ////
+
+    //! Get the origin position
+    CELER_FUNCTION Real3 const& origin() const { return origin_; }
+
+    //! Get involute parameters
+    CELER_FUNCTION real_type r_b() const { return r_b_; }
+    CELER_FUNCTION real_type a() const { return a_; }
+    CELER_FUNCTION real_type sign() const { return sign_; }
+
+    //! Get bounds of the involute
+    CELER_FUNCTION real_type tmin() const { return tmin_; }
+    CELER_FUNCTION real_type tmax() const { return tmax_; }
+
+    //! Get a view to the data for type-deleted storage
+    CELER_FUNCTION StorageSpan data() const { return {&origin_[0], 4}; }
+
+    // // Helper function to get the origin as a 3-vector
+    // Real3 calc_origin() const;
+
+    //// CALCULATION ////
+
+    // Determine the sense of the position relative to this surface
+    inline CELER_FUNCTION SignedSense calc_sense(Real3 const& pos) const;
+
+    // Calculate all possible straight-line intersections with this surface
+    inline CELER_FUNCTION Intersections calc_intersections(
+        Real3 const& pos, Real3 const& dir, SurfaceState on_surface) const;
+
+    // Calculate outward normal at a position
+    inline CELER_FUNCTION Real3 calc_normal(Real3 const& pos) const;
+
+  private:
+    // Location of the center of circle of involute
+    Real3 origin_;
+
+    // Involute parameters
+    real_type r_b_;
+    real_type a_;
+    real_type sign_;
+
+    // Bounds
+    real_type tmin_;
+    real_type tmax_;
+
+    //! Private default constructor for manual construction
+    Involute() = default;
```

Comment:
I don't think you'll need this once you remove the static constructor. It's kind of a rare use case.